packages/salesforcedx-vscode-core/src/commands/forceAuthDevHub.ts

```diff
@@ -37,6 +52,55 @@ export class ForceAuthDevHubExecutor extends SfdxCommandletExecutor<{}> {
     command.withArg('--setdefaultdevhubusername');
     return command.build();
   }
+
+  public async execute(response: ContinueResponse<any>): Promise<void> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async () => {
+      const globalDevHubName = await this.getDevNubHame(ConfigSource.Global);
+      if (isNullOrUndefined(globalDevHubName)) {
+        const localDevHubName = await this.getDevNubHame(ConfigSource.Local);
+        if (isNullOrUndefined(localDevHubName) === false) {
+          await this.setGlobal(String(localDevHubName));
+        }
+      }
+    });
+
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+  }
+
+  private async setGlobal(newUsername: string) {
+    const homeDirectory = require('os').homedir();
+    const configFileName = 'sfdx-config.json';
+
+    const globalConfig = await ConfigFile.create({
+      isGlobal: true,
+      rootFolder: homeDirectory,
+      filename: configFileName
+    });
+
+    globalConfig.set(defaultDevHubUserNameKey, newUsername);
+    await globalConfig.write();
+  }
+
+  public async getDevNubHame(source: ConfigSource.Global | ConfigSource.Local) {
```

Comment:
Can we use the `getDefaultDevHubUsernameOrAlias` in authInfo instead of this? The functionality seems similar

Reply:
That's a good idea and that's what I was doing initially

However, I found that the additional functionality provided by getDefaultDevHubUsernameOrAlias conflicts with our purposes here

More specifically, getDefaultDevHubUsernameOrAlias calls getConfigValue, and on the first occasion that we invoke getDefaultDevHubUsernameOrAlias, the argument ConfigSource.Global is provided. It then calls getConfigValue, and in the case that getConfigValue returns undefined, on line 63 in authInfo.ts, getDefaultDevHubUsernameOrAlias spins up an "await displayMessage", which prompts the user to "Authorize a Dev Hub" because there is no dev hub defined

Accordingly, in order to make use of getDefaultDevHubUsernameOrAlias, we would need to refactor it so that it stops prompting the user about there being no default dev hub, or we